net_transport.go

```diff
@@ -214,6 +246,17 @@ func (n *NetworkTransport) getPooledConn(target ServerAddress) *netConn {
 	return conn
 }
 
+// getConnFromAddressProvider returns a connection from the server address provider if available, or defaults to a connection using the target server address
+func (n *NetworkTransport) getConnFromAddressProvider(id ServerID, target ServerAddress) (*netConn, error) {
+	if n.serverAddressProvider != nil {
+		serverAddressOverride := n.serverAddressProvider.ServerAddr(id)
```

Comment:
I wonder if it's worthwhile to add an error return from the provider as well? Might make for a richer log message in this layer if something goes wrong in the lookup.

Reply:
`EncodePeer` doesn't have an error return, so it would need to be added there as well.

Reply:
I thought about that, but we would still fall back to the address in the method arg, so it means logging it here as opposed to logging it on the consul side, and you have more info like the datacenter on the consul side to log it there. 

It does make for a cleaner interface than assuming special return values like "", so that could be the argument for adding it.